lib/mode.coffee

```diff
@@ -0,0 +1,234 @@
+XRegExp = require 'xregexp'
+sb_exec = require 'sb-exec'
+path = require 'path'
+
+pattern = XRegExp('(?<file>[^\n\r]+):(?<from_line>\\d+):(?<from_col>\\d+):\\s*\
+  (?<to_line>\\d+):(?<to_col>\\d+)\\s+\
+  ((?<error>error|fatal error)|(?<warning>warning)|(?<info>note|help)):\\s+\
+  (?<message>.+?)[\n\r]+($|(?=[^\n\r]+:\\d+))', 's')
+
+parseOldMessages = (output, disabledWarnings) ->
+  elements = []
+  XRegExp.forEach output, pattern, (match) ->
+    if match.from_col == match.to_col
+      match.to_col = parseInt(match.to_col) + 1
```

Comment:
💡 Situations like this are where [`atom-linter::rangeFromLineNumber`](https://github.com/steelbrain/atom-linter/blob/v8.0.0/src/index.js#L16) is great as it will automatically determine a range highlighitng the next "word", _as defined by the language_. So you can feed it a single point and it will generate a better range than just the single character. It would require some refactoring though as it requires access to the `TextEditor` of the file being linted.